plugins/yaml/src/org/jetbrains/yaml/YAMLLanguageCodeStyleSettingsProvider.java

```diff
@@ -18,6 +15,9 @@
 import static com.intellij.psi.codeStyle.CodeStyleSettingsCustomizableOptions.getInstance;
 
 public class YAMLLanguageCodeStyleSettingsProvider extends LanguageCodeStyleSettingsProvider {
+
+  private static final YAMLCodeStyleSettingsProvider YAML_CODE_STYLE_SETTINGS_PROVIDER = new YAMLCodeStyleSettingsProvider();
```

Comment:
So there will be 2 instances: here and the instance is created by `plugin.xml`. Am I right?

Reply:
That is correct. Again, I don't quite understand the difference between the two providers, especially since `LanguageCodeStyleSettingsProvider` inherits from `CodeStyleSettingsProvider` (if I recall correctly). I don't know of any mechanism with which I could inject the instance from `plugin.xml` into this class, but to be honest I'm not extremely comfortable with this project.